device/cluster.cpp

```diff
@@ -1164,13 +1167,7 @@ void Cluster::write_device_memory(
 }
 
 void Cluster::read_device_memory(
-    void* mem_ptr,
-    tt_cxy_pair target,
-    std::uint32_t address,
-    std::uint32_t size_in_bytes,
-    const std::string& fallback_tlb) {
-    // Assume that mem_ptr has been allocated adequate memory on host when this function is called. Otherwise, this
-    // function will cause a segfault.
```

Comment:
I removed this because it's not true.  If there is inadequate memory where that points, you are lucky if your program segfaults.